src/common/decorators/subscription.decorator.ts

```diff
@@ -0,0 +1,49 @@
+import { IncomingHttpHeaders } from 'http';
+import {
+  ReturnTypeFunc,
+  Subscription,
+  SubscriptionOptions,
+} from '@nestjs/graphql';
+import { GraphQLResolveInfo } from 'graphql';
+import { AgentInfo } from '@src/core';
```

Comment:
Import should be more specific - this is pulling too much in, especially for a class that will be widely used.